diff --git a/src/runner.py b/src/runner.py
index 4bd672a..7af35a1 100644
--- a/src/runner.py
+++ b/src/runner.py
@@ -31,7 +31,7 @@ class Runner:
         if self.run_metadata_path is None or not self.run_metadata_path.exists():
             logger.info("run metadata not found. Generating...")
             print(f"Generating run metadata at {self.run_metadata_path}")
-            runner_metadata = RunnerAnalyzer(self.model_path)
+            runner_metadata = RunnerAnalyzer(self.model_path, self.metadata_path)
             self.run_metadata_path = runner_metadata.generate_metadata(save_path=self.run_metadata_path)
         
         with open(self.run_metadata_path, 'r') as f:
